neqo-bin/src/bin/client/main.rs

```diff
@@ -0,0 +1,541 @@
+                None
+            }
+        });
+
+    for ((host, port), mut urls) in urls_by_origin {
+        if args.resume && urls.len() < 2 {
+            eprintln!("Resumption to {host} cannot work without at least 2 URLs.");
+            exit(127);
+        }
+
+        let remote_addr = format!("{host}:{port}").to_socket_addrs()?.find(|addr| {
+            !matches!(
+                (addr, args.ipv4_only, args.ipv6_only),
+                (SocketAddr::V4(..), false, true) | (SocketAddr::V6(..), true, false)
+            )
+        });
+        let Some(remote_addr) = remote_addr else {
+            eprintln!("No compatible address found for: {host}");
+            exit(1);
+        };
+
+        let local_addr = match remote_addr {
+            SocketAddr::V4(..) => SocketAddr::new(IpAddr::V4(Ipv4Addr::from([0; 4])), 0),
+            SocketAddr::V6(..) => SocketAddr::new(IpAddr::V6(Ipv6Addr::from([0; 16])), 0),
+        };
+
+        let mut socket = udp::Socket::bind(local_addr)?;
+        let real_local = socket.local_addr().unwrap();
+        println!(
+            "{} Client connecting: {:?} -> {:?}",
+            if args.shared.use_old_http { "H9" } else { "H3" },
+            real_local,
+            remote_addr,
+        );
+
+        let hostname = format!("{host}");
+        let mut token: Option<ResumptionToken> = None;
+        let mut first = true;
+        while !urls.is_empty() {
+            let to_request = if (args.resume && first) || args.download_in_series {
+                urls.pop_front().into_iter().collect()
+            } else {
+                std::mem::take(&mut urls)
+            };
+
+            first = false;
+
+            let key_update = KeyUpdateState(args.key_update);
+
+            token = if args.shared.use_old_http {
+                let client =
+                    http09::create_client(&args, real_local, remote_addr, &hostname, token)
+                        .expect("failed to create client");
+
+                let handler = http09::Handler::new(to_request, &args, key_update);
+
+                Runner {
+                    args: &args,
+                    client,
+                    handler,
+                    local_addr: real_local,
+                    socket: &mut socket,
+                    timeout: None,
+                }
+                .run()
+                .await?
+            } else {
+                let client = http3::create_client(&args, real_local, remote_addr, &hostname, token)
+                    .expect("failed to create client");
+
+                let handler = http3::Handler::new(to_request, &args, key_update);
+
+                Runner {
+                    args: &args,
+                    client,
+                    handler,
+                    local_addr: real_local,
+                    socket: &mut socket,
+                    timeout: None,
+                }
+                .run()
+                .await?
+            };
```

Comment:
`http3` and `http09` expose the same methods (`create_client` and `Handler::new`) and can be _run_ with the same `Runner`.